scene/main/node.cpp

```diff
@@ -357,30 +361,32 @@ void Node::_move_child(Node *p_child, int p_index, bool p_ignore_end) {
 	// Specifying one place beyond the end
 	// means the same as moving to the last index
 	if (!p_ignore_end) { // p_ignore_end is a little hack to make back internal children work properly.
-		if (p_child->_is_internal_front()) {
-			if (p_index == data.internal_children_front) {
+		if (p_child->data.internal_mode == INTERNAL_MODE_FRONT) {
+			if (p_index == data.internal_children_front_cache) {
 				p_index--;
 			}
-		} else if (p_child->_is_internal_back()) {
-			if (p_index == data.children.size()) {
+		} else if (p_child->data.internal_mode == INTERNAL_MODE_BACK) {
+			if (p_index == (int)data.children_cache.size()) {
 				p_index--;
 			}
 		} else {
-			if (p_index == data.children.size() - data.internal_children_back) {
+			if (p_index == (int)data.children_cache.size() - data.internal_children_back_cache) {
 				p_index--;
 			}
 		}
 	}
 
-	if (p_child->data.index == p_index) {
+	int child_index = p_child->get_index();
+
+	if (child_index == p_index) {
 		return; //do nothing
 	}
 
-	int motion_from = MIN(p_index, p_child->data.index);
-	int motion_to = MAX(p_index, p_child->data.index);
+	int motion_from = MIN(p_index, child_index);
+	int motion_to = MAX(p_index, child_index);
 
-	data.children.remove_at(p_child->data.index);
-	data.children.insert(p_index, p_child);
+	data.children_cache.remove_at(child_index);
+	data.children_cache.insert(p_index, p_child);
```

Comment:
This should probably be made to be a single operation by adding relevant method in `LocalVector` (as currently it will needlessly shift elements after `MAX(p_index, child_index)` back and forth).

Reply:
Possibly a good idea, but beyond the scope of this PR, the original code did the same thing anyway.